oracle-plugin/src/main/java/co/cask/oracle/OracleSink.java

```diff
@@ -42,12 +52,30 @@ public OracleSink(OracleSinkConfig oracleSinkConfig) {
     this.oracleSinkConfig = oracleSinkConfig;
   }
 
+  @Override
+  public void transform(StructuredRecord input, Emitter<KeyValue<DBRecord, NullWritable>> emitter) throws Exception {
+    List<Schema.Field> outputFields = new ArrayList<>();
+    for (String column : columns) {
+      Schema.Field field = input.getSchema().getField(column);
+      Preconditions.checkNotNull(field, "Missing schema field for column '%s'", column);
+      outputFields.add(field);
+    }
+    StructuredRecord.Builder output = StructuredRecord.builder(
+      Schema.recordOf(input.getSchema().getRecordName(), outputFields));
+    for (String column : columns) {
```

Comment:
is this done in the base DB plugin? Is this copy being created in case the columns are a subset of the input record's schema? Does the OutputFormat not handle that case?